fractions/keycloak-microprofile-jwt/src/main/java/org/wildfly/swarm/keycloak/mpjwt/runtime/KeycloakMicroprofileJwtArchivePreparer.java

```diff
@@ -54,6 +63,63 @@ public void process() throws IOException {
         }
     }
 
+    private InputStream getKeycloakJsonFromSystemProperties() {
+        final String thorntailPrefix = "thorntail.keycloak.secure-deployments.[" + archive.getName() + "].";
+        final String swarmPrefix = "swarm.keycloak.secure-deployments.[" + archive.getName() + "].";
+        final String doubleQuote = "\"";
+        final String credentialsSecretProperty = "credentials.secret.value";
+
+        Map<String, String> kcAdapterSystemProps = new LinkedHashMap<>();
+        Properties systemProps = System.getProperties();
+        for (Enumeration<Object> en = systemProps.keys(); en.hasMoreElements();) {
+            String key = en.nextElement().toString();
+            if (key.startsWith(thorntailPrefix)) {
+                kcAdapterSystemProps.put(key.substring(thorntailPrefix.length()),
+                                         systemProps.getProperty(key));
+            } else if (key.startsWith(swarmPrefix)) {
+                kcAdapterSystemProps.put(key.substring(swarmPrefix.length()),
+                                         systemProps.getProperty(key));
+            }
+        }
+        InputStream is = null;
+        if (!kcAdapterSystemProps.isEmpty()) {
+            // Simple JSON builder tailored for the keycloak.json format
+            // which can contain simple string or boolean or integer values only
+            // with the only exception being a 'credentials' single entry map 
+            StringBuilder sb = new StringBuilder();
+            sb.append("{");
+            for (Map.Entry<String, String> entry : kcAdapterSystemProps.entrySet()) {
+                if (sb.length() > 1) {
+                    sb.append(",");
+                }
+                String key = entry.getKey();
+                boolean credentials = entry.getKey().equals(credentialsSecretProperty);
+                if (credentials) {
+                    key = "secret";
+                    sb.append("\"credentials\" : {");
+                }
+                sb.append(doubleQuote).append(key).append(doubleQuote);
+                sb.append(":");
+                String value = !credentials && (isBoolean(entry.getValue()) || isSimpleNumber(entry.getValue()))
+                    ? entry.getValue() : doubleQuote + entry.getValue() + doubleQuote;
+                sb.append(value);
```

Comment:
Can the value possibly contain something that would require escaping?

(In other words, I don't understand why people still try to assemble JSON by manually concatenating strings, it's _not_ easier than using a library.)

Reply:
Not in [keycloak.json](https://github.com/keycloak/keycloak/blob/master/adapters/oidc/adapter-core/src/test/resources/keycloak.json) as far as I'm aware (it is only simple numbers), this is why I'm not sure it is worth bringing some JSONP or whatever dependency just to build a basic JSON object.

Reply:
I.e, since it is a fairly specific task (for example, mapping between a `credentials.secret.value` and a JSON map property, etc), bringing in the library will not make it much simpler, I thought for a moment, OK, I will can write some generic code which accumulates all 'value' property and then generically convert them into JSON maps, etc. but then thought, was it really worth it ? :-), it is only a keycloak.json format which has to be supported

Reply:
Well, this fraction depends on Keycloak, so it already brings in Jackson Core :-)

Reply:
Sure, let me check this one as well, thanks

Reply:
@Ladicek, I've replaced the StringBuilder with Jackson JsonGenerator, it did not eliminate the code checking the simple number or boolean format since the input is not typed and is coming from the system properties, but it did help with removing that manual checks for commas, and the object start/end writes, so the code has become cleaner, thanks for suggesting it

Reply:
I will only replace Integer with Long

Reply:
Great, thanks!